js/react-tweek/src/createUseTweekValue.ts

```diff
@@ -60,7 +57,17 @@ export const createUseTweekValue = (
           }
         })
       );
-    }, [tweekRepository, keyPath, defaultValue]);
+    }, dependencies);
+
+    if (!isEqual(dependencies, storedDependencies.current)) {
+      storedDependencies.current = dependencies;
+
+      const newTweekValue = getTweekValue();
+      if (!isEqual(newTweekValue, tweekValue)) {
+        setTweekValue(newTweekValue);
```

Comment:
I think this is the same case as before the change. The difference now is that in the first render, before the call to `set`, the value returned is the correct value.

I don't think it's an issue, but it is interesting to try and think about a way to prevent this.
Although if the component is pure, the extra render won't occur

Reply:
Never change the state inside a render function. That's why we're using hooks.
And I still don't understand the problem, you're basically doing the same thing the useEffect did

Reply:
@nataly87s because useEffect runs after the render, in the case where the repository changes, the first render will return the old repository's value.
I think that if you'll look at the test I added it will make more sense.

As for not changing the state inside the render - I agree. But, we also should not change the state inside useEffect, because of the same reasons. (the double render problem occurs in both cases). Do you have an idea on how to fix this bug in a more elegant way?

Reply:
The value in the reducer state is incorrect in the render happening after the args change, but before the useEffect handler.

Reply:
@AssafKr Let me repeat that. **NEVER** change the state during the render stage!
you can use a bunch of different stuff for that, like `useMemo` `useEffect` `useAnyOtherHook` but never **ever** change the state inside the render function

Reply:
@nataly87s, we understand the problem here, obviously, it's not optimal. We're trying to find a way to synchronize the state of the reducer, while still allowing a render that happens after args change to present a correct value. (can fallback to default value)

Reply:
@AssafKr https://itnext.io/react-setstate-usage-and-gotchas-ac10b4e03d60

"**render**
Calling `setState` here makes your component a contender for producing infinite loops. `render` should remain pure and be used to conditionally switch between JSX fragments/child components based on state or props. Callbacks in render can be used to update state and then re-render based on the change.
If you find yourself having to write `setState` within `render`, you may want to rethink the design. In fact, this may be a perfect use-case to implement the state machines pattern."